### PR TITLE
Replica cache fix

### PR DESCRIPTION
**Pull request by hohice**
### What changes were proposed in this pull request?

update replicat type cache when update type 

### Why are the changes needed?

need it to modify replica type 

### Will break the compatibility? How if so?

No

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

extand replica with readony type then modify to full

### Checklist

- [x] I've run the tests to see all new and existing tests pass.
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [x] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

**Comment by simonjoylet** [comment]
please describe the background information for this pr.

File: src/storage/ob_pg_storage.cpp [hunk 1]
```diff
@@ -2335,8 +2335,8 @@ int ObPGStorage::set_pg_replica_type(const ObReplicaType& replica_type, const bo
           storage->get_partition_store().set_replica_type(replica_type);
         }
       }
-      switch_meta_(next_meta_ptr);
```
Comment (simonjoylet) [inline src/storage/ob_pg_storage.cpp:2338]: why delete this?

Comment (hohice) [reply src/storage/ob_pg_storage.cpp:2338]: miss when reopen PR. will add it

**Comment by hohice** [comment]
backgroud:
if without update , when extend replica with readonly then modify to full will get below
```shell
[op1@VMSALI053111 log]$ cat observer.log* |grep -i YB420A1FCECD-0005D81F45C00B7E
[2022-02-17 14:58:17.347898] WARN  [COMMON] get_max_iops (ob_io_benchmark.cpp:1047) [4254][1808][YB420A1FCECD-0005D81F45C00B7E] [lt=108] [dc=0] The ObIOBenchmark has not been inited, (ret=-4006)
[2022-02-17 14:58:17.366679] INFO  [SHARE.PT] ob_partition_location_cache.cpp:3148 [4274][1848][YB420A1FCECD-0005D81F45C00B7E] [lt=99] [dc=0] add_update_task success(task={table_id:{value:1100611139403897, first:1001, second:121}, partition_id:0, add_timestamp:1645081097366650, cluster_id:1025, type:0}, leader_inactive_async_update=false, now=1645081097366650, location={table_id:{value:1100611139403897, first:1001, second:121}, partition_id:0, partition_cnt:0, pg_key:{tid:18446744073709551615, partition_id:-1, part_idx:268435455, subpart_idx:268435455}, replica_locations:[{server:"10.29.196.74:2882", role:1, sql_port:2881, replica_type:0, reserved:0, property:{memstore_percent_:100}}, {server:"10.60.44.84:2882", role:2, sql_port:2881, replica_type:16, reserved:0, property:{memstore_percent_:100}}, {server:"10.60.45.26:2882", role:2, sql_port:2881, replica_type:0, reserved:0, property:{memstore_percent_:100}}, {server:"10.60.52.75:2882", role:2, sql_port:2881, replica_type:0, reserved:0, property:{memstore_percent_:100}}], renew_time:1645005134373622, sql_renew_time:1645004939109081, is_mark_fail:false}, is_nonblock=false)
[2022-02-17 14:58:17.366697] INFO  [SHARE.PT] ob_partition_location_cache.cpp:2371 [4274][1848][YB420A1FCECD-0005D81F45C00B7E] [lt=10] [dc=0] add_update_task success(task={table_id:{value:1100611139403897, first:1001, second:121}, partition_id:0, add_timestamp:1645081097366692, cluster_id:1025, type:1})
[2022-02-17 14:58:17.372108] ERROR [STORAGE] update_rows (ob_pg_storage.cpp:3417) [4254][1808][YB420A1FCECD-0005D81F45C00B7E] [lt=101] [dc=0] replica is not writable(ret=-5081, this={this:0x7f98fa985540, pkey:{tid:1100611139403897, partition_id:0, part_cnt:0}, meta:0x7f9894beff70, log_seq_num:0, last_freeze_ts:1645006978396144}) BACKTRACE:0x9786d2e 0x96d9aa1 0x226092f 0x226057b 0x2260342 0x7e7eb9b 0x7ce8a7f 0x81e1fb9 0x7b6445a 0x517d8d3 0x5109cca 0x5109513 0x38ef56e 0x54ec13e 0x5481880 0x542a9f5 0x98ad35c 0x90a956f 0x9089efb 0x90a81cf 0x9087e04 0x9088327 0x2bf4284 0x95718e5 0x9570572 0x956d09f
[2022-02-17 14:58:17.374440] INFO  [STORAGE.TRANS] ob_memtable_context.cpp:1491 [3989][1278][YB420A1FCECD-0005D81F45C00B7E] [lt=5] [dc=0] memtable handle rollback to successfuly(sql_no=1, is_replay=false, need_write_log=false, max_durable_log_ts=0, *this=alloc_type=0 ctx_descriptor=13074652 trans_start_time=1645081097367044 min_table_version=216 max_table_version=216 is_safe_read=false has_read_relocated_row=false read_snapshot=1645081097361218 start_version=-1 trans_version=9223372036854775807 commit_version=0 stmt_start_time=1645081097372073 abs_expired_time=1645082097277475 stmt_timeout=999905402 abs_lock_wait_timeout=1645082097247475 row_purge_version=0 lock_wait_start_ts=0 trx_lock_timeout=-1 end_code=0 is_readonly=false ref=0 pkey={tid:1100611139403897, partition_id:0, part_cnt:0} trans_id={hash:7050723871125267607, inc:7451888, addr:"10.60.45.26:2882", t:1645081097366029} data_relocated=0 relocate_cnt=0 truncate_cnt=0 trans_mem_total_size=61 callback_alloc_count=2 callback_free_count=0 callback_mem_used=480 checksum_log_ts=0, has_calc_checksum=false, checksum=0, checksum_log_ts=0)
[2022-02-17 14:58:17.347898] WARN  [COMMON] get_max_iops (ob_io_benchmark.cpp:1047) [4254][1808][YB420A1FCECD-0005D81F45C00B7E] [lt=108] [dc=0] The ObIOBenchmark has not been inited, (ret=-4006)
[2022-02-17 14:58:17.372108] ERROR [STORAGE] update_rows (ob_pg_storage.cpp:3417) [4254][1808][YB420A1FCECD-0005D81F45C00B7E] [lt=101] [dc=0] replica is not writable(ret=-5081, this={this:0x7f98fa985540, pkey:{tid:1100611139403897, partition_id:0, part_cnt:0}, meta:0x7f9894beff70, log_seq_num:0, last_freeze_ts:1645006978396144}) BACKTRACE:0x9786d2e 0x96d9aa1 0x226092f 0x226057b 0x2260342 0x7e7eb9b 0x7ce8a7f 0x81e1fb9 0x7b6445a 0x517d8d3 0x5109cca 0x5109513 0x38ef56e 0x54ec13e 0x5481880 0x542a9f5 0x98ad35c 0x90a956f 0x9089efb 0x90a81cf 0x9087e04 0x9088327 0x2bf4284 0x95718e5 0x9570572 0x956d09f
```

**Review by simonjoylet — APPROVED** [review]

**Pull request merged**